dGame/dComponents/VendorComponent.h

```diff
@@ -2,98 +2,66 @@
 #ifndef VENDORCOMPONENT_H
 #define VENDORCOMPONENT_H
 
+#include <functional>
 #include "CDClientManager.h"
 #include "Component.h"
 #include "Entity.h"
 #include "GameMessages.h"
 #include "RakNetTypes.h"
 #include "eReplicaComponentType.h"
 
-/**
- * A component for vendor NPCs. A vendor sells items to the player.
- */
+struct SoldItem {
+	SoldItem(const LOT lot, const int32_t sortPriority) {
+		this->lot = lot;
+		this->sortPriority = sortPriority;
+	};
+	LOT lot = 0;
+	int32_t sortPriority = 0;
+};
+
 class VendorComponent : public Component {
 public:
-	static const eReplicaComponentType ComponentType = eReplicaComponentType::VENDOR;
-
+	inline static const eReplicaComponentType ComponentType = eReplicaComponentType::VENDOR;
 	VendorComponent(Entity* parent);
-	~VendorComponent() override;
-
 	void Serialize(RakNet::BitStream* outBitStream, bool bIsInitialUpdate, unsigned int& flags);
-
 	void OnUse(Entity* originator) override;
-
-	/**
-	 * Gets the buy scaler
-	 * @return the buy scaler
-	 */
-	float GetBuyScalar() const;
-
-	/**
-	 * Sets the buy scalar.
-	 * @param value the new value.
-	 */
-	void SetBuyScalar(float value);
-
-	/**
-	 * Gets the buy scaler
-	 * @return the buy scaler
-	 */
-	float GetSellScalar() const;
-
-	/**
-	 * Sets the sell scalar.
-	 * @param value the new value.
-	 */
-	void SetSellScalar(float value);
-
-	/**
-	 * True if the NPC LOT is 13800, the only NPC with a crafting station.
-	 */
-	bool HasCraftingStation();
-
-	/**
-	 * Gets the list if items the vendor sells.
-	 * @return the list of items.
-	 */
-	std::map<LOT, int>& GetInventory();
-
-	/**
-	 * Refresh the inventory of this vendor.
-	 */
 	void RefreshInventory(bool isCreation = false);
-
-	/**
-	 * Called on startup of vendor to setup the variables for the component.
-	 */
 	void SetupConstants();
-
-	bool SellsItem(const LOT item) const;
+	float GetBuyScalar() const { return m_BuyScalar; }
+	float GetSellScalar() const { return m_SellScalar; }
+	void SetBuyScalar(const float value) { m_BuyScalar = value; }
+	void SetSellScalar(const float value) { m_SellScalar = value; }
+
+	std::vector<SoldItem>& GetInventory() { return m_Inventory; }
+
+	void SetHasMultiCostItems(const bool hasMultiCostItems) {
+		if (m_HasMultiCostItems == hasMultiCostItems) return;
+		m_HasMultiCostItems = hasMultiCostItems;
+		m_DirtyVendor = true;
+	}
+
+	void SetHasStandardCostItems(const bool hasStandardCostItems) {
+		if (m_HasStandardCostItems == hasStandardCostItems) return;
+		m_HasStandardCostItems = hasStandardCostItems;
+		m_DirtyVendor = true;
+	}
+
+	bool SellsItem(const LOT item) const {
+		return std::count_if(m_Inventory.begin(), m_Inventory.end(), [item](const SoldItem& lhs) {
+			return lhs.lot == item;
+			}) > 0;
+	}
```

Comment:
For moving these checkers to header files, we should generally only do so if it is small like the above `SetHasStandardCostItems`.  including the entire functional header in this header just to have this in the header file is a bit heavy of a cost